### PR TITLE
[ONNX] Add ReverseSequence operator

### PR DESCRIPTION
**Pull request by mbencer**

### CODE REVIEW

File: src/ngraph/frontend/onnx_import/CMakeLists.txt [hunk 1]
```diff
@@ -155,6 +155,8 @@ add_library(onnx_import STATIC
         op/relu.hpp
         op/reshape.cpp
         op/reshape.hpp
+		op/reverse_sequence.cpp
```
Comment (silee2) [inline src/ngraph/frontend/onnx_import/CMakeLists.txt]: fix indentation

Comment (mbencer) [reply src/ngraph/frontend/onnx_import/CMakeLists.txt]: Done. I had incorrect settings of IDE ;)

File: src/ngraph/frontend/onnx_import/op/reverse_sequence.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <memory>
+#include <vector>
```
[the remark this answers is not in the document]
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/op/reverse_sequence.cpp]: You are right, memory and vector are not needed in this .cpp

File: test/onnx/onnx_import.in.cpp [hunk 3]
```diff
@@ -1577,3 +1577,41 @@ NGRAPH_TEST(onnx_${BACKEND_NAME}, model_eye_like)
 
     test_case.run();
 }
+
+NGRAPH_TEST(onnx_${BACKEND_NAME}, reverse_sequence_0_batch_1)
```
[the remark this answers is not in the document]
Comment (mbencer) [reply test/onnx/onnx_import.in.cpp]: Done

File: test/models/onnx/reverse_sequence_time_0_batch_1.prototxt [hunk 4]
```diff
@@ -0,0 +1,69 @@
+ir_version: 3
+producer_name: "nGraph ONNX Importer"
+graph {
+  node {
+    input: "x"
+	  input: "sequence_lengths"
```
[the remark this answers is not in the document]
Comment (mbencer) [reply test/models/onnx/reverse_sequence_time_0_batch_1.prototxt]: Good idea. I used `onnx_prototxt_converter.py` to fix indentation issues

File: test/models/onnx/reverse_sequence_time_1_batch_0.prototxt [hunk 5]
```diff
@@ -0,0 +1,69 @@
+ir_version: 3
+producer_name: "nGraph ONNX Importer"
+graph {
+  node {
+    input: "x"
+	  input: "sequence_lengths"
```
[the remark this answers is not in the document]
Comment (mbencer) [reply test/models/onnx/reverse_sequence_time_1_batch_0.prototxt]: Issues resolved

File: src/ngraph/frontend/onnx_import/op/reverse_sequence.h [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            {
+                NodeVector reverse_sequence(const Node& node);
+
+            } // namespace set_1
+
+        } //namespace op
+
+    } // namespace onnx_import
+
+} // namespace ngraph
```
Comment (tomdol) [inline src/ngraph/frontend/onnx_import/op/reverse_sequence.h]: New line missing at the end of this file

File: src/ngraph/frontend/onnx_import/op/reverse_sequence.cpp [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/reverse_sequence.hpp"
+#include "core/node.hpp"
+#include "ngraph/node.hpp"
+
+using namespace ngraph::op;
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/op/reverse_sequence.cpp]: It looks like you still use explicit full qualified name for classes, thus this is unnecessary.

File: src/ngraph/frontend/onnx_import/op/reverse_sequence.cpp [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+            namespace set_1
+            {
+                NodeVector reverse_sequence(const Node& node)
+                {
+                    const auto data = node.get_ng_inputs().at(0);
+                    const auto sequence_lengths = node.get_ng_inputs().at(1);
+                    const auto batch_axis = node.get_attribute_value<int64_t>("batch_axis", 1);
+                    const auto time_axis = node.get_attribute_value<int64_t>("time_axis", 0);
+
+                    return {std::make_shared<ngraph::op::ReverseSequence>(
```
Comment (arogowie-intel) [inline src/ngraph/frontend/onnx_import/op/reverse_sequence.cpp]: Please add header `<memory>`.

File: src/ngraph/frontend/onnx_import/op/reverse_sequence.h (hunk 6 above)
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/op/reverse_sequence.h]: Done

File: src/ngraph/frontend/onnx_import/op/reverse_sequence.cpp (hunk 8 above)
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/op/reverse_sequence.cpp]: Done

File: src/ngraph/frontend/onnx_import/op/reverse_sequence.cpp (hunk 7 above)
Comment (mbencer) [reply src/ngraph/frontend/onnx_import/op/reverse_sequence.cpp]: You are right. I removed unused declared napespace

**Pull request merged**